### PR TITLE
Add `hideOnClose` flag in `<DialogTrigger />`

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Describe changes

This PR add ability to specify the way how we'd like to hide a dialog: by unmounting or by hiding in css

##### Checklist

Before taking this PR from the draft, please, make sure you have done the following:

- [x] Pipeline is passed
- [x] Tests are added (including unit tests and stories in the storybook)
- [x] Tests are passed successfully
- [x] If you're adding a new component/new props, add stories that describe how this component/prop works
- [x] Changeset(s) is(are) added
- [x] You have passed the threshold of the library size
- [x] Commit message follows [commit guidelines](https://github.com/cube-js/cube-ui-kit/blob/main/CONTRIBUTING.md)

Closes:  N/A 

#

### CODE REVIEW

File: src/components/overlays/Dialog/DialogTrigger.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface CubeDialogTriggerProps {
   mobileViewport?: number;
   /** The style map for the overlay **/
   styles?: Styles;
+  closeBehavior?: 'hide' | 'remove';
```
Comment (tenphi) [inline src/components/overlays/Dialog/DialogTrigger.tsx:44]: I'm pretty sure it should be called `destroy`, not `remove`.
image

Comment (MrFlashAccount) [reply src/components/overlays/Dialog/DialogTrigger.tsx:44]: @tenphi I've read the answer from `ChatGPT` and it looks like it stands on my side ))

Btw: https://developer.mozilla.org/en-US/docs/Web/API/Element/remove, using `remove` matches with web platform better than `destroy`

Comment (MrFlashAccount) [reply src/components/overlays/Dialog/DialogTrigger.tsx]: Renamed `hideBehavior` -> `hideOnClose`

File: src/components/overlays/Modal/Underlay.tsx [hunk 2]
```diff
@@ -20,10 +20,14 @@ const UnderlayElement = tasty({
...
     },
+    visibility: {
+      '': 'hidden',
+      open: 'visible',
+    },
     fill: '#dark.30',
     overflow: 'hidden',
     transition:
-      'transform .25s ease-in-out, opacity .25s linear, visibility 0ms linear',
+      'transform .25s ease-in-out, opacity .25s linear, visibility 0.13s linear',
```
Comment (tenphi) [inline src/components/overlays/Modal/Underlay.tsx:30]: You can use the default `@transition` custom property for `120ms`.

**Review by tenphi — APPROVED** [review]

**Review by tenphi — APPROVED** [review]

**Pull request merged**